pstoreds/addr_book.go

```diff
@@ -56,17 +63,29 @@ func (mgr *dsAddrBook) Stop() {
 	mgr.ttlManager.cancel()
 }
 
-func peerAddressKey(p *peer.ID, addr *ma.Multiaddr) (ds.Key, error) {
-	hash, err := mh.Sum((*addr).Bytes(), mh.MURMUR3, -1)
-	if err != nil {
-		return ds.Key{}, nil
+func keysAndAddrs(p peer.ID, addrs []ma.Multiaddr) ([]ds.Key, []ma.Multiaddr, error) {
```

Comment:
This function now takes all the peers and multiaddrs, discards the invalid ones and converts them to ds.Keys, also returning only the valid multiaddrs. Before, each method was doing this.